tests/tasks/async_queries_tests.py

```diff
@@ -202,3 +203,29 @@ def test_soft_timeout_load_explore_json_into_cache(
                 ensure_user_is_set.side_effect = SoftTimeLimitExceeded()
                 load_explore_json_into_cache(job_metadata, form_data)
             ensure_user_is_set.assert_called_once_with(user.id, "error", errors=errors)
+
+    def test_ensure_user_is_set(self):
+        g_user_is_set = hasattr(g, "user")
+        original_g_user = g.user if g_user_is_set else None
+
+        if g_user_is_set:
+            del g.user
+
+        self.assertFalse(hasattr(g, "user"))
+        ensure_user_is_set(1)
+        self.assertTrue(hasattr(g, "user"))
+        self.assertFalse(g.user.is_anonymous)
+        self.assertEqual("1", g.user.get_id())
+
+        del g.user
+
+        self.assertFalse(hasattr(g, "user"))
+        ensure_user_is_set(None)
+        self.assertTrue(hasattr(g, "user"))
+        self.assertTrue(g.user.is_anonymous)
+        self.assertEqual(None, g.user.get_id())
+
+        if g_user_is_set:
+            g.user = original_g_user
+        else:
+            del g.user
```

Comment:
In reference to the above comment, should we add a test case for cases where `g.user` is set and see what `g.user` is after calling `ensure_user_is_set`?